rpc/src/module/test.rs

```diff
@@ -62,6 +471,9 @@ pub(crate) struct IntegrationTestRpcImpl {
 
 impl IntegrationTestRpc for IntegrationTestRpcImpl {
     fn process_block_without_verify(&self, data: Block, broadcast: bool) -> Result<Option<H256>> {
+        // check if on dev network
+        self.check_dev_chain()?;
```

Comment:
These checks should only need to be done once at rpc server build time, not on every request.  (https://github.com/nervosnetwork/ckb/blob/develop/rpc/src/service_builder.rs#L145-L164)